core/src/main/scala/cats/syntax/option.scala

```diff
@@ -390,7 +390,7 @@ final class OptionOps[A](private val oa: Option[A]) extends AnyVal {
 }
 
 object OptionOps {
-  final private[syntax] class LiftToPartiallyApplied[F[_], A](oa: Option[A]) {
+  final class LiftToPartiallyApplied[F[_], A](oa: Option[A]) {
```

Comment:
Why isn't it an AnyVal? Seems like it could be.

I saw a comment that it could break binary compatibility but I think MiMa would catch it. Can we not try if we are making this public?

Reply:
Sure! Just pushed a change 👍

Reply:
Unfortunately, it seems that MiMa is not happy with that change :(

Reply:
reverted:c